libevm/VM.cpp

```diff
@@ -1096,7 +1492,9 @@ void VM::interpretCases()
 #if EVM_HACK_DUP_64
 			*(uint64_t*)m_SPP = *(uint64_t*)(m_SP + n);
 #else
-			m_SPP[0] = m_SP[n];
+			// the stack slot being copied into may no longer hold a u256
+			// so we construct a new one in the memory, rather than assign
+			new(m_SPP) u256(m_SP[n]);
```

Comment:
So do I understand this correctly, that this is done to avoid copy assignment operator interpreting the data as `u256` and somehow corrupting it, since it can be not `u256` at all?

But this placement new still calls the copy constructor, and copy constructor will still interpret it as `u256`.

Maybe we should `memcpy` or `std:copy` the bytes here? Whis is against the rules since uint256 is not a POD...
But I don't see how placement new helps here